src/ProgramBlockEditor.js

```diff
@@ -570,14 +650,92 @@ class ProgramBlockEditor extends React.Component<ProgramBlockEditorProps, Progra
+                            aria-disabled={this.props.editingDisabled}
+                            role='button'
+                            tabIndex='0'
+                            value='left'
+                            onKeyDown={!this.props.editingDisabled ? this.handleKeyDownCharacterPosition : undefined}
+                            onClick={!this.props.editingDisabled ? this.handleClickCharacterPosition : undefined} />
                         <div
                             className='ProgramBlockEditor__character-column-character-container'
                             role='img'
                             aria-label={this.getThemedCharacterAriaLabel()}>
                             {this.getThemedCharacter()}
                         </div>
-                    </h3>
+                        <MovePositionRight
+                            className={characterPositionButtonClassName}
+                            aria-label={this.props.intl.formatMessage({id:'ProgramBlockEditor.editPosition.moveRight'})}
+                            aria-disabled={this.props.editingDisabled}
+                            role='button'
+                            tabIndex='0'
+                            value='right'
+                            onKeyDown={!this.props.editingDisabled ? this.handleKeyDownCharacterPosition : undefined}
+                            onClick={!this.props.editingDisabled ? this.handleClickCharacterPosition : undefined} />
+                    </div>
+                    <div className='ProgramBlockEditor__character-move-position-bottom'>
+                        <MovePositionDown
+                            className={characterPositionButtonClassName}
+                            aria-label={this.props.intl.formatMessage({id:'ProgramBlockEditor.editPosition.moveDown'})}
+                            aria-disabled={this.props.editingDisabled}
+                            role='button'
+                            tabIndex='0'
+                            value='down'
+                            onKeyDown={!this.props.editingDisabled ? this.handleKeyDownCharacterPosition : undefined}
+                            onClick={!this.props.editingDisabled ? this.handleClickCharacterPosition : undefined} />
+                    </div>
+                    <div className='ProgramBlockEditor__character-move-position-coordinate'>
+                        <input
+                            className={characterPositionTextInputClassName}
+                            aria-label={this.props.intl.formatMessage({id:'ProgramBlockEditor.editPosition.columnPosition'})}
+                            aria-disabled={this.props.editingDisabled}
+                            type='text'
+                            value={this.state.characterColumnLabel}
+                            onChange={!this.props.editingDisabled ? this.handleChangeXPos : undefined}
```

Comment:
Given that the change doesn't take effect until you blur the control, I'm not sure that `onChange` is adding anything.  In fact you might be attempting to update the position twice.

As discussed elsewhere, we should have support for hitting enter as well.  I think `onBlur` and a small `onKeyDown` handler would be good, unless you have a concrete reason for using `onChange` related to a particular browser.

Reply:
onChange handler is responsible for updating values represented on the text box UI, but that doesn't actually change values of CharacterState. We don't want to apply changes as users change text box values, because that will send potentially invalid values for xPos and yPos in CharacterState. OnBlur will communicate with CharacterState state in App and update xPos and yPos in CharacterState after input validation.